src/storage/engine/mod.rs

```diff
@@ -69,9 +69,12 @@ pub enum Modify {
     DeleteRange(CfName, Key, Key),
 }
 
-pub trait Engine: Send + Debug {
+pub trait Engine: Send + Debug + Clone + Sized + 'static {
+    type IteratorType: Iterator;
```

Comment:
why not using I and S for simplicity.

Reply:
Because normally we tend to write full name for associated types. 
See `Iterator::Type` in [Ref1](https://doc.rust-lang.org/book/second-edition/ch19-03-advanced-traits.html) and `Future::Item` / `Future::Error` in [Ref2](https://docs.rs/futures/0.2.1/futures/trait.Future.html)

Reply:
Also, I intensionally added "Type" suffix to distinguish it with `Iterator` and `Snapshot` (these two names are already taken as a Trait)

Reply:
IMO, this associated type name is a little long, and Type is also strange here, It must be a Type, we don't need to re-declare it again.

Reply:
So now `IterateType` is renamed to `Iter` and `SnapshotType` is renamed to `Snap` for simplicity. Also `<EngineRocksdb as Engine>::SnapshotType` becomes `RocksSnapshot` now.